tests/rptest/utils/audit_schemas.py

```diff
@@ -0,0 +1,403 @@
+            False
+        },
+        'resources': {
+            'type': 'object',
+            'properties': {
+                'name': {
+                    'type': 'string'
+                },
+                'type': {
+                    'type': 'string'
+                }
+            },
+            'required': ['name', 'type'],
+            'additionalProperties': False
+        },
+        'src_endpoint': ENDPOINT_SCHEMA,
+        'status_id': {
+            'type': 'number',
+            'enum': [0, 1, 2, 99]
+        },
+        'unmapped': {
+            'type': 'object',
+            'properties': {
+                'shard_id': {
+                    'type': 'number'
+                },
+                'authorization_metadata': AUTH_METADATA_SCHEMA
+            },
+            'required': ['shard_id'],
+            'additionalProperties': False
+        }
+    },
+    'required': [
+        'activity_id', 'actor', 'api', 'dst_endpoint', 'src_endpoint',
+        'status_id', 'unmapped'
+    ],
+    'additionalProperties':
+    False
+}
+
+APPLICATION_ACTIVITY_SCHEMA = {
+    'type': 'object',
+    'properties': {
+        'activity_id': {
+            'type': 'number',
+            'enum': [0, 1, 2, 3, 4, 99]
+        },
+        'app': PRODUCT_SCHEMA
+    },
+    'required': ['activity_id', 'app'],
+    'additionalProperties': False
+}
+
+
+# A proper schema is an application based schema + the fields that exist
+# in the OCSF base schema.
+def _create_schema(schema):
+    return {
+        'type': 'object',
+        'properties': schema['properties'] | OCSF_BASE_SCHEMA['properties'],
+        'required': schema['required'] + OCSF_BASE_SCHEMA['required'],
+        'additionalProperties': False
+    }
+
+
+# All schemas outside of this list are nested within the supported structures
+# and are not expected to be observed as-is in the audit log
+VALID_AUDIT_SCHEMAS = [
+    _create_schema(schema)
+    for schema in [API_ACTIVITY_SCHEMA, APPLICATION_ACTIVITY_SCHEMA]
+]
+
+
+class AuditSchemeValidationException(Exception):
+    def __init__(self, excs):
+        failures = [str(exc) for exc in excs]
+        self.message = '\n'.join(failures)
+
+
+def validate_audit_contents(message):
+    """
+    This method verifies the validity of the contents of a message that
+    has already passed the schema check.
+    """
+    if 'count' not in message:
+        return 'start_time' not in message and 'end_time' not in message
+
+    if message['count'] <= 1:
+        return False
+    if 'start_time' not in message:
+        return False
+    if 'end_time' not in message:
+        return False
+    return True
+
+
+def validate_audit_schema(message):
+    """
+    Verify that a given message adheres to one of the supported audit log
+    record formats and if one of those checks pass, that the fields within
+    the records are themselves valid.
+    """
+    excs = []
+    for schema in VALID_AUDIT_SCHEMAS:
```

Comment:
I wanted one exception message in the case `validate_audit_contents` failed, otherwise you'd see 2 exception messages when the second would be representing a case thats not true, since a record that adheres to the schema is guaranteed to pass at least one of the schemas in the `VALID_AUDIT_SCHEMAS` list and fail all of the others

Reply:
I think if you added a `continue` after `execs.append` in your snippet it would be functionally equivalent